apps/projects/app/App.js

```diff
@@ -57,247 +59,176 @@ const getTabs = ({ repoCount }) => {
   return tabs
 }
 
-class App extends React.PureComponent {
-  static propTypes = {
-    api: PropTypes.object, // is not required, since it comes async
-    repos: PropTypes.arrayOf(PropTypes.object),
-    bountySettings: PropTypes.object,
-    displayMenuButton: PropTypes.bool,
-    client: PropTypes.object,
-    issues: PropTypes.array,
-    tokens: PropTypes.array,
-    github: PropTypes.shape({
-      status: PropTypes.oneOf([
-        STATUS.AUTHENTICATED,
-        STATUS.FAILED,
-        STATUS.INITIAL,
-      ]).isRequired,
-      token: PropTypes.string,
-      event: PropTypes.string,
-    }),
-  }
-
-  constructor(props) {
-    super(props)
-    this.state = {
-      repos: [],
-      activeIndex: { tabIndex: 0, tabData: {} },
-      githubLoading: false,
-      issueDetail: false,
-      panel: null,
-      panelProps: null,
-    }
-  }
-
-  componentDidMount() {
-    /**
-     * Acting as the redirect target it looks up for 'code' URL param on component mount
-     * if it detects the code then sends to the opener window
-     * via postMessage with 'popup' as origin and close the window (usually a popup)
-     */
+const App = () => {
+  const { api, appState } = useAragonApi()
+  const [ activeIndex, setActiveIndex ] = useState(
+    { tabIndex: 0, tabData: {} }
```

Comment:
"Data" was added to Tabs index to enable moving between Tabs from within a Tab and pass information at the same time. Specifically clicking on a Project Card results in Tab changing to Issues and issues from the chosen Projects to be displayed. One way of avoiding it (and operating on index only, without passing any data between Tabs) in this specific case would involve manipulating filtering, so Overview tab (Project Card) can change Tab to Issues and at the same time add appropriate filtering - in other words Overview would update Tab Index in its parent, App, and update filtering in Issues Tab. THe way it works now is that Overview passes repo Id to filter by, and Issues in `componentWillMount` configures filtering appropriately. Is it the best way? Probably not, but it's working and refactoring it could be done as post-shipping phase. Unless you think there's an obvious and simple way to refactor it now?